File: GA8-220501096-AA1-EV01/NotePlusTaller.API/Models/Persona.cs
namespace NotePlusTaller.API.Models;

public class Persona
{
    public int IDPersona { get; set; }
    public int IDNacionalidad { get; set; }
    public string TipoDocumento { get; set; } = string.Empty;
    public string NumeroDocumento { get; set; } = string.Empty;
    public string PrimerNombre { get; set; } = string.Empty;
    public string? SegundoNombre { get; set; }
    public string PrimerApellido { get; set; } = string.Empty;
    public string? SegundoApellido { get; set; }
    public DateOnly FechaNacimiento { get; set; }
    public string? TelefonoFijo { get; set; }
    public string? Celular { get; set; }
    public string CorreoElectronico { get; set; } = string.Empty;
    public string? DireccionResidencia { get; set; }
    public bool Estado { get; set; }

    public Nacionalidad Nacionalidad { get; set; } = null!;
    public Usuario? Usuario { get; set; }
}

File: GA8-220501096-AA2-EV02/app/src/main/java/com/example/noteplusadmin/data/entities/Persona.kt
package com.example.noteplusadmin.data.entities

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

@Entity(
    tableName = "persona",
    foreignKeys = [
        ForeignKey(
            entity = Nacionalidad::class,
            parentColumns = ["idNacionalidad"],
            childColumns = ["idNacionalidad"],
            onDelete = ForeignKey.RESTRICT
        )
    ]
)
data class Persona(
    @PrimaryKey(autoGenerate = true) val idPersona: Int = 0,
    @ColumnInfo(index = true) val idNacionalidad: Int,
    val tipoDocumento: String,
    val numeroDocumento: String,
    val primerNombre: String,
    val segundoNombre: String? = null,
    val primerApellido: String,
    val segundoApellido: String? = null,
    val fechaNacimiento: String,
    val telefonoFijo: String? = null,
    val celular: String,
    val correoElectronico: String,
    val direccionResidencia: String,
    val estado: String
)
